pkg/unidling/controller/controller.go

```diff
@@ -0,0 +1,323 @@
+	if targetScalablesStr, hasTargetScalables := targetEndpoints.Annotations[unidlingapi.UnidleTargetAnnotation]; hasTargetScalables {
+		if err = json.Unmarshal([]byte(targetScalablesStr), &targetScalables); err != nil {
+			// retrying here won't help, we're just stuck as idled since we can't parse the idled scalables list
+			return false, fmt.Errorf("unable to unmarshal target scalable references: %v", err)
+		}
+	} else {
+		glog.V(4).Infof("Service %s/%s had no scalables to unidle", info.Namespace, info.Name)
+		targetScalables = []unidlingapi.RecordedScaleReference{}
+	}
+
+	targetScalablesSet := make(map[unidlingapi.RecordedScaleReference]struct{}, len(targetScalables))
+	for _, v := range targetScalables {
+		targetScalablesSet[v] = struct{}{}
+	}
+
+	scales := c.scaleNamespacer.Scales(info.Namespace)
+	for _, scalableRef := range targetScalables {
+		var scale *kextapi.Scale
+		scale, err = scales.Get(scalableRef.Kind, scalableRef.Name)
+		if err != nil {
+			if errors.IsNotFound(err) {
+				utilruntime.HandleError(fmt.Errorf("%s %q does not exist, removing from list of scalables while unidling service %s/%s: %v", scalableRef.Kind, scalableRef.Name, info.Namespace, info.Name, err))
+				delete(targetScalablesSet, scalableRef)
+			} else {
+				utilruntime.HandleError(fmt.Errorf("Unable to get scale for %s %q while unidling service %s/%s, will try again later: %v", scalableRef.Kind, scalableRef.Name, info.Namespace, info.Name, err))
+			}
+			continue
+		}
+
+		if scale.Spec.Replicas > 0 {
+			glog.V(4).Infof("%s %q is not idle, skipping while unidling service %s/%s", scalableRef.Kind, scalableRef.Name, info.Namespace, info.Name)
+			continue
+		}
+
+		scale.Spec.Replicas = scalableRef.Replicas
+
+		if _, err = scales.Update(scalableRef.Kind, scale); err != nil {
+			if errors.IsNotFound(err) {
+				utilruntime.HandleError(fmt.Errorf("%s %q does not exist, removing from list of scalables while unidling service %s/%s: %v", scalableRef.Kind, scalableRef.Name, info.Namespace, info.Name, err))
+				delete(targetScalablesSet, scalableRef)
+			} else {
+				utilruntime.HandleError(fmt.Errorf("Unable to scale up %s %q while unidling service %s/%s: %v", scalableRef.Kind, scalableRef.Name, info.Namespace, info.Name, err))
+			}
+			continue
+		} else {
+			glog.V(4).Infof("Scaled up %s %q while unidling service %s/%s", scalableRef.Kind, scalableRef.Name, info.Namespace, info.Name)
+		}
+
+		delete(targetScalablesSet, scalableRef)
+	}
+
+	newAnnotationList := make([]unidlingapi.RecordedScaleReference, 0, len(targetScalablesSet))
+	for k := range targetScalablesSet {
+		newAnnotationList = append(newAnnotationList, k)
+	}
+
+	if len(newAnnotationList) == 0 {
+		delete(targetEndpoints.Annotations, unidlingapi.UnidleTargetAnnotation)
+		delete(targetEndpoints.Annotations, unidlingapi.IdledAtAnnotation)
+	} else {
+		var newAnnotationBytes []byte
+		newAnnotationBytes, err = json.Marshal(newAnnotationList)
```

Comment:
Make sure you update to use proper conversions and serialization.  This ends up being part of the API.